include/ginkgo/core/base/index_set.hpp

```diff
@@ -100,6 +100,27 @@ class IndexSet : public EnablePolymorphicObject<IndexSet<IndexType>> {
         : EnablePolymorphicObject<IndexSet>(std::move(exec))
     {}
 
+    /**
+     * Creates an index set on the specified executor from the initializer list.
+     *
+     * @param exec  the Executor where the index set data will be allocated
+     * @param init_list  the indices that the index set should hold in an
+     *                   initializer_list.
+     * @param is_sorted  a parameter that specifies if the indices array is
+     *                   sorted or not. `true` if sorted.
+     */
+    IndexSet(std::shared_ptr<const gko::Executor> executor,
```

Comment:
```suggestion
    explicit IndexSet(std::shared_ptr<const gko::Executor> executor,
```
same for the other constructors